common/git-hooks/pre-commit

```diff
@@ -6,4 +6,4 @@
 # Invoke the "rush prettier" custom command to reformat files whenever they
 # are committed. The command is defined in common/config/rush/command-line.json
 # and uses the "rush-prettier" autoinstaller.
-node common/scripts/install-run-rush.js prettier || exit $?
+prettier --write --ignore-unknown . || exit $?

```

Comment:
This runs prettier on every file in the repository (less those excluded by prettierignore), which might work in a small repo (of which rushstack is reasonably so), but won't really scale well.

Reply:
That's what I noticed too. I think what we want is to run prettier on changed files only. So I guess we need to retrieve the changed files from staged files?

Reply:
Unfortunately the prettier CLI doesn't support passing paths from stdin, and if you specify exact paths on the command line, the prettierignore seems to get bypassed. Ultimately this means we have to wrap it in a node script that calls the API, at which point it is probably easier to just have a package in the repository that gets installed as part of normal `rush install` and has the dependency on `prettier`. Then the hook can invoke the node script directly and error if `prettier` has not yet been installed.